src/core/net/srp_client.cpp

```diff
@@ -415,6 +426,22 @@ void Client::HandleNotifierEvents(Events aEvents)
         ProcessAutoStart();
     }
 #endif
+
+    if (mHostInfo.IsAutoAddressEnabled())
+    {
+        Events::Flags eventFlags = (kEventIp6AddressAdded | kEventIp6AddressRemoved);
+
+        if (mAutoHostAddressAddedMeshLocal)
+        {
+            eventFlags |= kEventThreadMeshLocalAddrChanged;
+        }
+
+        if (aEvents.ContainsAny(eventFlags))
+        {
+            IgnoreError(UpdateHostInfoStateOnAddressChange());
+            UpdateState();
```

Comment:
Not necessarily for this PR, but we should consider whether we should include a random delay before sending an SRP Update message. The current logic can lead to networks with large numbers of devices causing a flood of SRP Update messages whenever the OMR prefix changes.

Reply:
Good idea. 

We actually already start the timer when we enter `kStateToUpdate` (from `Client::SetState()`) but it is fixed short duration of  `kUpdateTxDelay` (which is 10 msec by default). We can add a random jitter to it. Will add in a future/new PR.

Reply:
Shouldn't this be part of the Thread Specification?

Reply:
+1 this should be required in Thread spec.